### PR TITLE
Send a message in Hub discussion after Parquet conversion

### LINKED ISSUE #1270: Send a notification to the dataset owner on parquet files creation
Once the parquet files have been created, we could send a notification to the dataset maintainer, to let them know, with the goal to increase the usage of these files.

Note: an idea by @julien-c when we started working on parquet conversion was to do it through the PR mechanism. It would have helped in that aspect, since the dataset maintainers would have received the notification as with any other PR.

### PR DESCRIPTION
**Pull request by severo**
fixes #1270.

Implements https://github.com/huggingface/datasets-server/issues/1270#issuecomment-1726384167

### CODE REVIEW

File: jobs/cache_maintenance/tests/test_discussions.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ({"a/b"}, [{"a/b"}]),
+        ({"a"}, [set()]),
+        ({"a/b/c"}, [set()]),
+        ({"a/b", "a/b"}, [{"a/b"}]),
+        ({"a/b", "a/c"}, [{"a/b"}, {"a/c"}]),
+        ({"a/b", "b/b"}, [{"a/b", "b/b"}]),
+        ({"a/b", "b"}, [{"a/b"}]),
+    ],
+)
+def test_limit_to_one_dataset_per_namespace(datasets: set[str], valid_expected_datasets: list[set[str]]) -> None:
```
Comment (AndreaFrancis) [inline jobs/cache_maintenance/tests/test_discussions.py:25]: nit: Can you add test for post_messages function (maybe in another PR)?

**Review by AndreaFrancis — APPROVED** [review]
Great!

**Review by albertvillanova — APPROVED** [review]
Thanks.

File: jobs/cache_maintenance/tests/test_discussions.py (hunk 1 above)
Comment (severo) [reply jobs/cache_maintenance/tests/test_discussions.py:29]: it was not a "nit" at all... I found and fixed innumerable bugs.

**Comment by severo** [comment]
@julien-c @stevhliu @osanseviero @davanstrien: if you have recommendations for the content or format of the notification.

Capture d’écran 2023-09-20 à 23 44 14

**Comment by julien-c** [comment]
`Parquet-converter (BOT) ` can you link the user account that will be used here? (we already use it for commits no?)

**Comment by julien-c** [comment]
My 2 cents: the second message in your screenshot looks cleaner/simpler than the 1st one.

**Comment by severo** [comment]
> My 2 cents: the second message in your screenshot looks cleaner/simpler than the 1st one.

Yes. The second one is for any update after the first message. I can replace the first message with a generic one that tells the discussion is for Datasets Server notifications and that closing it equals unsubscribing.

**Comment by davanstrien** [comment]
> > My 2 cents: the second message in your screenshot looks cleaner/simpler than the 1st one.
> 
> Yes. The second one is for any update after the first message. I can replace the first message with a generic one that tells the discussion is for Datasets Server notifications and that closing it equals unsubscribing.

I think it would be good in the first message to give some motivation for why someone should want a parquet version. I think this could a link to an external doc too. Maybe modify the sentence about the docs to something like: 

> There are some advantages associated with having a version of your dataset avaialble in the parquet format. You can learn more about these in the docs.

**Comment by severo** [comment]
> Parquet-converter (BOT) can you link the user account that will be used here? (we already use it for commits no?)

It's https://huggingface.co/parquet-converter.

**Comment by severo** [comment]
I updated a bit, based on your comments:

Capture d’écran 2023-09-21 à 18 53 12

**Comment by severo** [comment]
Let's try it! It will run tonight for the first time. We will look at the logs in kibana to get an idea of the volume of messages

**Comment by julien-c** [comment]
Also if you can share some links to Discussions that would be helpful (including some with responses from repo owners)

BTW maybe would be nice to have a call to action to suggest feedback directly in the discussion

**Comment by severo** [comment]
> BTW maybe would be nice to have a call to action to suggest feedback directly in the discussion

See https://github.com/huggingface/datasets-server/pull/1846

**Comment by severo** [comment]
Seems like it didn't run last night:

https://kibana.elastic.huggingface.tech/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-24h%2Fh,to:now))&_a=(columns:!(message,kubernetes.namespace),filters:!(),index:de38ff80-ac19-11ec-bb45-ad141ad1c5f8,interval:auto,query:(language:kuery,query:'message:%20%22Post%20messages%20in%20Hub%20discussion%20to%20notify%20about%20parquet%20conversion%22'),sort:!(!('@timestamp',desc)))

**Comment by severo** [comment]
For some reason, the [cron jobs](https://github.com/huggingface/datasets-server/tree/main/chart/templates/cron-jobs) logs don't seem to be stored by prometheus. cc @rtrompier @AndreaFrancis

**Comment by rtrompier** [comment]
Are you talking about logs or metrics ? 
metrics -> prometheus
logs -> logstash -> elasticsearch

**Comment by severo** [comment]
Oh yes, sorry: logs -> logstash -> elasticsearch

**Comment by rtrompier** [comment]
Kibana link shared on slack.

**Comment by julien-c** [comment]
so, did it work?

**Comment by severo** [comment]
Seems like it didn't do anything (no logs; and no ["community activity"](https://github.com/huggingface/datasets-server/pull/1824#issuecomment-1735138168))

**Comment by severo** [comment]
OK, I finally got the logs (thanks @rtrompier!)

```
WARNING:root:Wrong action mode selected, supported actions are ['backfill', 'collect-cache-metrics', 'collect-queue-metrics', 'delete-indexes', 'skip'].
```

I have to fix a detail. Nothing was done yesterday

**Comment by severo** [comment]
> 294 messages posted (total: 298 datasets): 294 discussions have been opened. 0 messages have been dismissed because the discussion had been closed. 4 errors.

**Comment by severo** [comment]
The errors:
```
Failed to post a message for shunk031/JGLUE: 403 Client Error: Forbidden for url: https://huggingface.co/api/datasets/shunk031/JGLUE/discussions?p=0 (Request ID: Root=1-651372e4-18b9a4df17a09b0c6a977b9a;793ed897-a533-4101-887f-5e993ae28615)
Failed to post a message for JosephusCheung/user-access-report-JosephusCheung: 403 Client Error: Forbidden for url: https://huggingface.co/api/datasets/JosephusCheung/user-access-report-JosephusCheung/discussions?p=0 (Request ID: Root=1-651372f8-2939eb1529b877e62d31cd3f;982c7a32-1c51-46c6-8283-f94c181591d2)
Failed to post a message for ccmusic-database/music_genre: 403 Client Error: Forbidden for url: https://huggingface.co/api/datasets/ccmusic-database/music_genre/discussions?p=0 (Request ID: Root=1-65137313-16baab573ec3b74901c1be0e;99580111-d051-48ef-8c18-175208f831c8)
Failed to post a message for rukkuhru/LoRAData: 403 Client Error: Forbidden for url: https://huggingface.co/api/datasets/rukkuhru/LoRAData/discussions?p=0 (Request ID: Root=1-65137319-4e5cdd93182b0ec5038ee0b2;588a6092-50da-4576-a51d-b20f3bbba045)
```

**Comment by severo** [comment]
The community tab was disabled for these 4 datasets, so the errors are normal (even if I didn't expect them, because I was not aware of the feature).

https://huggingface.co/datasets/shunk031/JGLUE

Capture d’écran 2023-09-27 à 09 15 09
Capture d’écran 2023-09-27 à 09 16 22

**Comment by severo** [comment]
Discussion opened on one of my datasets:

https://huggingface.co/datasets/severo/trending-repos/discussions/1

Capture d’écran 2023-09-27 à 09 21 54

**Comment by severo** [comment]
Four users closed the discussion to stop receiving messages.
 
Capture d’écran 2023-09-27 à 09 27 12

**Comment by julien-c** [comment]
will be interesting to measure response / reaction rates.

Similar to what @davanstrien did recently with automated PRs measurement, even if here it's different as it's a Discussion not a PR. (cc @LysandreJik too)

**Comment by severo** [comment]
```
INFO: 2023-09-28 00:11:08,161 - root - 271 messages posted (total: 272 datasets): 218 discussions have been opened. 0 messages have been dismissed because the discussion had been closed. 1 errors.
```

- 3 discussions closed by the dataset owner
- 0 feedback

**Comment by severo** [comment]
> INFO: 2023-09-29 00:10:52,473 - root - 252 messages posted (total: 255 datasets): 188 discussions have been opened. 1 messages have been dismissed because the discussion had been closed. 2 errors.

- 2 errors due to disabled community tab
- 3 discussions closed by the dataset owner
- 0 feedback

I'll stop posting daily updates here. I'll comment if somebody writes a feedback, or if something breaks

**Comment by severo** [comment]
Some succinct feedback comments 😀:

- "Cool": https://huggingface.co/datasets/facebook/winoground/discussions/5#65230f48ef81ce01a4839002
- "👍": https://huggingface.co/datasets/Javiai/failures-3D-print/discussions/1#6520a1fe971e5c1370e1ccfe
- "cool": https://huggingface.co/datasets/julien-c/persistent-space-dataset/discussions/3#652022569334173c62391caa

Nobody complained.

**Comment by severo** [comment]
Oups, it's not what we intended:


https://huggingface.co/datasets/Yukang/LongAlpaca-12k/discussions/1#6524bdaee0823fbfef9bd5f0

> Many thanks for your help. I have merged these files into the main branch.

**Comment by severo** [comment]
Another positive feedback:

Capture d’écran 2023-10-10 à 11 12 37

**Comment by severo** [comment]
Feedback: https://huggingface.co/datasets/mcipriano/stackoverflow-kubernetes-questions/discussions/2#6525250e4471e38b4cbf415f

> The dataset is already in parquet format and is present in the data folder. The csv file was only put in to provide this version as well

The docs don't state that the Parquet datasets are not converted but simply "copied". I will propose a PR for this page: https://huggingface.co/docs/datasets-server/parquet, along the lines of https://github.com/ml6team/fondant/pull/491

**Comment by severo** [comment]
https://huggingface.co/datasets/junaav/flower512/discussions/1#6527928c5b3582bcb8760f7e

> this is my own dataset and please do not share anywhere, thanks

**Comment by severo** [comment]
https://huggingface.co/datasets/open-source-metrics/datasets-dependents/discussions/1#652d85cdcc4609d6438a0b02

> [@severo](https://huggingface.co/severo) this is maybe a bit much in terms of pings haha

cc @julien-c

**Comment by severo** [comment]
Cool:

> Oh cool, didn't even realize this was a thing thank you!

https://huggingface.co/datasets/EarthnDusk/Creative-Embeddings-OLD/discussions/1#652f354e61ce8120848d3deb

**Comment by julien-c** [comment]
don't hesitate to reply from your personal account btw. To engage conversation etc

**Comment by severo** [comment]
Yes? I'll try to do more (not just emojis)

For https://huggingface.co/datasets/open-source-metrics/datasets-dependents/discussions/1#652d85cdcc4609d6438a0b02, maybe there is an issue with org datasets -> maybe some people want to see the notification, while other don't.

**Comment by julien-c** [comment]
i think maybe at some point we want the ability for users to unsubscribe from notifs from certain bots

(can you create a moon-landing issue for this actually?)

**Comment by severo** [comment]
Good idea -> https://github.com/huggingface/moon-landing/issues/7807 (internal)

**Comment by severo** [comment]
https://github.com/huggingface/moon-landing/issues/7778#issuecomment-1765247955

> Right now messages often get swamped by safetenors, parquet conversion, other events that don't usually carry actionable items for me.

**Comment by severo** [comment]
Even the bots don't want notifications anymore 📦 

Capture d’écran 2023-10-20 à 10 44 54

**Comment by severo** [comment]
see follow-up PR that reduces the amount of spam: https://github.com/huggingface/datasets-server/pull/2024

**Comment by julien-c** [comment]
so there'll be only one post now?

**Comment by severo** [comment]
Yes

**Pull request merged**